llvm/test/tools/sycl-post-link/emit_exported_symbols.ll

```diff
@@ -0,0 +1,129 @@
+; RUN: FileCheck %s -input-file=%t.per_module.files_1.prop --check-prefixes CHECK-PERMODULE-1-PROP
+; RUN: FileCheck %s -input-file=%t.per_module.files_2.prop --check-prefixes CHECK-KERNELONLY-PROP
+;
+; Per-kernel split
+; RUN: sycl-post-link -symbols -split=kernel -emit-exported-symbols -S %s -o %t.per_kernel.files.table
+; RUN: FileCheck %s -input-file=%t.per_kernel.files.table --check-prefixes CHECK-PERKERNEL-TABLE
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_0.prop --check-prefixes CHECK-PERKERNEL-0-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_1.prop --check-prefixes CHECK-PERKERNEL-1-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_2.prop --check-prefixes CHECK-PERKERNEL-2-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_3.prop --check-prefixes CHECK-KERNELONLY-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_4.prop --check-prefixes CHECK-KERNELONLY-PROP
+
+target triple = "spir64-unknown-unknown"
+
+define dso_local spir_kernel void @SpirKernel1(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_kernel void @SpirKernel2(float %arg1) #2 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc1(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc2(i32 %arg1, i32 %arg2) #1 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc3(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @NotExportedSpirFunc1(float %arg1) {
+entry:
+  ret void
+}
+
+attributes #0 = { "sycl-module-id"="a.cpp" }
+attributes #1 = { "sycl-module-id"="b.cpp" }
+attributes #2 = { "sycl-module-id"="c.cpp" }
+
+; Global scope
+; CHECK-GLOBAL-PROP: [SYCL/exported symbols]
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc1
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc2
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc3
+; CHECK-GLOBAL-PROP-NOT: SpirKernel1
+; CHECK-GLOBAL-PROP-NOT: NotExportedSpirFunc1
+
+; CHECK-GLOBAL-TABLE: [Code|Properties|Symbols]
+; CHECK-GLOBAL-TABLE-NEXT: {{.*}}global.files_0.prop
+; CHECK-GLOBAL-TABLE-EMPTY:
+
+; Per-module split
+; CHECK-PERMODULE-0-PROP: [SYCL/exported symbols]
+; CHECK-PERMODULE-0-PROP-NEXT: ExportedSpirFunc1
+; CHECK-PERMODULE-0-PROP-NEXT: ExportedSpirFunc3
+; CHECK-PERMODULE-0-PROP-NOT: ExportedSpirFunc2
+; CHECK-PERMODULE-0-PROP-NOT: SpirKernel1
+; CHECK-PERMODULE-0-PROP-NOT: NotExportedSpirFunc1
+
+; CHECK-PERMODULE-1-PROP: [SYCL/exported symbols]
+; CHECK-PERMODULE-1-PROP-NEXT: ExportedSpirFunc2
+; CHECK-PERMODULE-1-PROP-NOT: ExportedSpirFunc1
+; CHECK-PERMODULE-1-PROP-NOT: ExportedSpirFunc3
+; CHECK-PERMODULE-1-PROP-NOT: SpirKernel1
+; CHECK-PERMODULE-1-PROP-NOT: NotExportedSpirFunc1
+
+; CHECK-PERMODULE-TABLE: [Code|Properties|Symbols]
+; CHECK-PERMODULE-TABLE-NEXT: {{.*}}per_module.files_0.prop
+; CHECK-PERMODULE-TABLE-NEXT: {{.*}}per_module.files_1.prop
+; CHECK-PERMODULE-TABLE-NEXT: {{.*}}per_module.files_2.prop
+; CHECK-PERMODULE-TABLE-EMPTY:
+
```

Comment:
I think checks for the table content could be omitted. The only thing they check is that device code split has been performed correctly, but we already have separate tests for that and here (for simplicity) we can just assume that it works as expected

Reply:
Good point. I have removed the table file checks.